non-oss/pharos_pro/addons/kontena-storage/addon.rb

```diff
@@ -122,24 +127,94 @@ def set_defaults
     }
   end
 
+  # @return [String]
+  def rook_version
+    self.class.version.split('+').first
+  end
+
+  # @param cluster [K8s::Resource]
+  # @param ceph_version [String]
+  def upgrade_from_08(cluster, ceph_version)
+    storage_stack = kube_stack(
+      cluster: cluster.to_h.deep_transform_keys(&:to_s),
+      rook_version: rook_version
+    )
+    logger.info "Applying upgrade ..."
+    storage_stack.apply(kube_client, prune: false)
+
+    logger.info "Waiting for new operator ..."
+    wait_mgr_upgrade(ceph_version)
+
+    logger.info "Cleaning up old configurations ..."
+    storage_stack.prune(kube_client, keep_resources: true)
+    remove_old_mgr(ceph_version)
+  end
+
+  # Wait new ceph mgr replica set
+  #
+  # @param ceph_version [String]
+  def wait_mgr_upgrade(ceph_version)
+    rs_client = kube_client.api('extensions/v1beta1').resource('replicasets', namespace: 'kontena-storage')
+    upgraded = false
+    while !upgraded do
+      upgraded = rs_client.list(labelSelector: 'app=rook-ceph-mgr').any? { |rs|
+        rs.spec.template.spec.containers.first.image.include?("ceph/ceph:v#{ceph_version}")
+      }
```

Comment:
Maybe a bit of sleeping as this is now a busyloop

Reply:
Fixed